### ISSUE TITLE
Memory usage of direct ethernet, qubit server

### ISSUE THREAD

**Issue by rb25** [martinisgroup/servers#161]
At Vince, we've been running into some slow datataking issues, related to pushing the memory of redheat.

Below are screenshots from the processes for three cases:
-running RB, after running RB for over a the whole night
   mem use direct ethernet: 315 MB
   mem use qubitserver: 1 GB
   mem use labrad: 743 MB
-stopped all scans (no measuring going on for a few minutes), didn't restart any server
   mem use direct ethernet: 435 MB
   mem use qubitserver: 535 MB
   mem use labrad: 681 MB
-restarted servers: direct ethernet, qubit sequencer and fpgaserver, and running RB
   mem use direct ethernet: 184 MB
   mem use qubitserver: 549 MB
   mem use labrad: 642 MB

There might be two things going on at the same time:
- the qubitserver doesn't have a memory leak, but it's garbage collection only acts when it is doing nothing for a long time, and the memory use keeps growing
- the direct ethernet server may have a memory leak

Below are the screenshots:
(running RB, after running RB for over a the whole night)
![qubitserver-1gb-runningrb](https://cloud.githubusercontent.com/assets/11541945/7730478/741f4ec8-fecf-11e4-800f-206e9e368f51.png)

(stopped all scans (no measuring going on for a few minutes), didn't restart any server)
![process-runningnothing](https://cloud.githubusercontent.com/assets/11541945/7730485/7963b252-fecf-11e4-95b4-392c8c2b28e5.png)

(restarted servers: direct ethernet, qubit sequencer and fpgaserver, and running RB)
![process-afterrestarting_qubitsequencer-directethernet_fgpaserver_runningrb](https://cloud.githubusercontent.com/assets/11541945/7730487/7cc343d6-fecf-11e4-9014-d122d39caac1.png)

And the memory use after RB pushing the limits (no scan was running at this time):

![memuse-nothingrunning](https://cloud.githubusercontent.com/assets/11541945/7730494/87d20d66-fecf-11e4-99f4-9a8a9e7893e6.png)

**Comment by maffoo** [comment]
@rb25 can I get a block of time on the fridge to do some heap profiling on the servers? I'd like to see if we can figure out whether there really is a memory leak and/or how we should tune the garbage collector.

**Comment by rb25** [comment]
Totally. If you want the whole of today, I'm fine with that. Let me know if you need any help.

**Comment by maffoo** [comment]
I did some heap and cpu profiling on the server processes on redheat and didn't see any evidence of a memory leak. I did find that the default maximum heap size is 1G on 64-bit windows with our version of java, so that's why the processes were climbing up to about that level in some cases. I've set the max heap size for the server down to 600M which should prevent java from eating up all 4 gigs on the box and slowing things down. We can monitor and adjust as needed.
